src/validation.cairo

```diff
@@ -50,7 +50,13 @@ impl TransactionValidatorImpl of TransactionValidator {
 }
 
 fn block_hash(self: @ChainState, block: @Block, merkle_root: u256) -> Result<u256, ByteArray> {
-    // TODO: implement
+    let header = block.header;
+
+    let mut header_data = ArrayTrait::<u32>::new();
+    header_data.append(*header.version);
+    header_data.append(*header.time);
+    header_data.append(*header.bits);
+    header_data.append(*header.nonce);
     Result::Ok(0)
```

Comment:
You need to use `compute_sha256_u32_array` here. Use `double_sha256` just as an inspiration.

Reply:
I'm using that function `compute_sha256_u32_array` for merkle and previous block but since they are different types I'm getting errors.